### PR TITLE
Automated cherry pick of #304: E2E framework

### PR DESCRIPTION
**Pull request by wongma7**
Cherry pick of #304 on release-1.22.

#304: Simple E2E runner

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
NONE
```

### CODE REVIEW

**Comment by wongma7** [comment]
I used https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh . conflicts were in go.mod/go.sum, of course for 1.22 branch we should stick to 1.22 dependencies.

File: tests/e2e/go.mod [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	github.com/google/uuid v1.1.4 // indirect
+	github.com/imdario/mergo v0.3.11 // indirect
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.10.3
+	gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b // indirect
+	k8s.io/api v0.22.0
+	k8s.io/apimachinery v0.22.0
+	k8s.io/client-go v9.0.0+incompatible
+	k8s.io/kubectl v0.17.2 // indirect
+	k8s.io/kubernetes v1.22.0
```
Comment (wongma7) [inline tests/e2e/go.mod:15]: i find/replaced 23.0 with 22.0. probably could use a patch release, don't think it much matters

**Comment by wongma7** [comment]
since https://github.com/kubernetes/test-infra/pull/25460 is still pending, here is manual run in this 1.22 branch:
```

• [SLOW TEST:116.002 seconds]
[cloud-provider-aws-e2e] loadbalancer
/home/ANT.AMAZON.COM/mattwon/go/src/k8s.io/cloud-provider-aws/tests/e2e/loadbalancer.go:25
  should configure the loadbalancer based on annotations
  /home/ANT.AMAZON.COM/mattwon/go/src/k8s.io/cloud-provider-aws/tests/e2e/loadbalancer.go:42
------------------------------

Ran 1 of 1 Specs in 116.003 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

File: tests/e2e/go.mod (hunk 1 above)
Comment (nckturner) [reply tests/e2e/go.mod:15]: perhaps not since its just test framework dependencies

**Comment by wongma7** [comment]
the bot thinks release note  is needed cuz for k/k all cherry-picks must have one as justification for cherry-pick in first place? Not sure how to trick it otherwise.

**Comment by wongma7** [comment]
NVM adding 
```release-note
NONE
```
fixed it I think.

**Pull request merged**